sklearn/tree/_tree.pyx

```diff
@@ -115,8 +122,9 @@ cdef class TreeBuilder:
         if (sample_weight is not None and
             (sample_weight.dtype != DOUBLE or
             not sample_weight.flags.contiguous)):
-                sample_weight = np.asarray(sample_weight, dtype=DOUBLE,
-                                           order="C")
+                # sample weight should be contiguous array
+                sample_weight = np.ascontiguousarray(sample_weight,
```

Comment:
`asarray(..., order="C")` is the same as `ascontiguousarray`. Since the code does the same thing, I prefer to revert to keep the `git blame` .

Reply:
Done